src/ActivityRepository.js

```diff
@@ -0,0 +1,101 @@
+class ActivityRepository {
+  constructor(activityData, userData) {
+    this.activityData = activityData;
+    this.userData = userData;
+  }
+
+  getUserActivityData(id) {
+    let singleUserData = this.activityData.filter(dataPoint => {
+      return dataPoint.userID === id;
+    })
+    return singleUserData;
+  }
+
+  getGeneralUserData(id) {
+    let generalUserData = this.userData.find(dataPoint => {
+      return dataPoint.id === id;
+    })
+    return generalUserData;
+  }
+
+  getActiveMinutes(id, date) {
+    let singleUserData = this.getUserActivityData(id);
+    let dataByDate = singleUserData.find(dataPoint => {
+      return dataPoint.date === date;
+    })
+    return dataByDate.minutesActive;
+  }
+
+  verifyStepAchievement(id, date) {
+    let singleUserActivity = this.getUserActivityData(id);
+    let generalUserData = this.getGeneralUserData(id);
+    let dailyStepGoal = generalUserData.dailyStepGoal;
+    let dataByDate = singleUserActivity.find(dataPoint => {
+      return dataPoint.date === date;
+    })
+    return dataByDate.numSteps >= generalUserData.dailyStepGoal ? true : false;
```

Comment:
I know what this line is doing, but not how it's happening. Can you explain what is happening throughout  `return dataByDate.numSteps >= generalUserData.dailyStepGoal ? true : false;`

Reply:
yes! so, line 36 involves a ternary.  It essentially works like this:

return `condition to be evaluated` ? `outcome if true` : `outcome if false`

In this case, we want to know if the `numSteps` value is greater than or equal to `dailyStepGoal`, each of which belong as properties to two different sets of data.

If `numSteps` is greater than or equal to `dailyStepGoal`, then `true` is returned.  Otherwise, if the condition evaluates to false, then `false` is returned.

Does that make more sense?

Reply:
Yes! thank you!